File: JavaVersion/app/src/main/java/com/thesis/javaversion/Movie.java
package com.thesis.javaversion;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.io.Serializable;
import java.util.ArrayList;

@Entity(tableName = "movie_table")
public class Movie implements Serializable {

    @PrimaryKey(autoGenerate = true)
    private int ID; // Unique

    @ColumnInfo(name = "title")
    private String title;
    @ColumnInfo(name = "plot")
    private String plot;
    @ColumnInfo(name = "release_date")
    private String releaseDate;
    @ColumnInfo(name = "genre")
    private ArrayList<String> genre;
    @ColumnInfo(name = "age_rated")
    private String ageRated;
    @ColumnInfo(name = "score")
    private int score;
    @ColumnInfo(name = "img_url")
    private String imgUrl;

    public int getID() {
        return ID;
    }

    public void setID(int ID) {
        this.ID = ID;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getPlot() {
        return plot;
    }

    public void setPlot(String plot) {
        this.plot = plot;
    }

    public String getReleaseDate() {
        return releaseDate;
    }

    public void setReleaseDate(String releaseDate) {
        this.releaseDate = releaseDate;
    }

    public ArrayList<String> getGenre() {
        return genre;
    }

    public void setGenre(ArrayList<String> genre) {
        this.genre = genre;
    }

    public String getAgeRated() {
        return ageRated;
    }

    public void setAgeRated(String ageRated) {
        this.ageRated = ageRated;
    }

    public int getScore() {
        return score;
    }

    public void setScore(int score) {
        this.score = score;
    }

    public String getImgUrl() {
        return imgUrl;
    }

    public void setImgUrl(String imgUrl) {
        this.imgUrl = imgUrl;
    }
}

File: KotlinVersion/app/src/main/java/com/thesis/kotlinversion/Movie.kt
package com.thesis.kotlinversion

import androidx.room.PrimaryKey
import androidx.room.ColumnInfo
import androidx.room.Entity
import java.io.Serializable
import java.util.ArrayList

@Entity(tableName = "movie_table")
class Movie : Serializable {
    @PrimaryKey(autoGenerate = true)
    var iD = 0

    @ColumnInfo(name = "title")
    var title: String? = null

    @ColumnInfo(name = "plot")
    var plot: String? = null

    @ColumnInfo(name = "release_date")
    var releaseDate: String? = null

    @ColumnInfo(name = "genre")
    var genre: ArrayList<String>? = null

    @ColumnInfo(name = "age_rated")
    var ageRated: String? = null

    @ColumnInfo(name = "score")
    var score = 0

    @ColumnInfo(name = "img_url")
    var imgUrl: String? = null
}